.github/workflows/gradle.yml

```diff
@@ -1,17 +1,98 @@
 name: Gradle build
 
-on: [push]
+on: [ push ]
 
 jobs:
-  build:
+  build-ubuntu:
+    runs-on: ubuntu-20.04
 
-    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Install Chrome
+        run: |
+          sudo apt install -y libappindicator1 fonts-liberation
+          wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+          sudo dpkg -i google-chrome*.deb
+      - name: Cache gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            **/.gradle
+            **/build
+          key: gradle_6_6_1
+          restore-keys: gradle_6_6_1
+
+      - name: Cache konan
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.konan/dependencies
+            ~/.konan/kotlin-native-prebuilt-linux-1.4.20
+          key: ${{ runner.os }}-konan-1.4.20
+          restore-keys: ${{ runner.os }}-konan-1.4.20
+      - name: Build with Gradle
+        run: ./gradlew -Dorg.gradle.daemon=false --build-cache build
+
+  build-osx:
+    runs-on: macos-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Cache gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            **/.gradle
+            **/build
+          key: gradle_6_6_1
+          restore-keys: gradle_6_6_1
+
+      - name: Cache konan
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.konan/dependencies
+            ~/.konan/kotlin-native-prebuilt-macos-1.4.20
+          key: ${{ runner.os }}-konan-1.4.20
+          restore-keys: ${{ runner.os }}-konan-1.4.20
+      - name: Build with Gradle
+        run: sudo ./gradlew -Dorg.gradle.daemon=false --build-cache build
+
+  build-windows:
```

Comment:
Platform builds will take a lot of time. I suggest to run only one platform automatically and others on-demand. We can do that for releases/platform sensitive things.

Reply:
The build time is incresead only by 5 minutes because Windows runners are pretty slow.